pkg/kubelet/kubelet.go

```diff
@@ -2124,6 +2178,43 @@ func (kl *Kubelet) HandlePodAdditions(pods []*v1.Pod) {
 	}
 }
 
+func (kl *Kubelet) checkPodResourcesResize(pod *v1.Pod) {
+	podResized := false
+	for _, container := range pod.Spec.Containers {
+		if len(diff.ObjectDiff(container.ResourcesAllocated, container.Resources.Requests)) > 0 {
+			podResized = true
+			break
+		}
+	}
+	if !podResized {
+		return
+	}
+
+	var otherActivePods []*v1.Pod
+	activePods := kl.GetActivePods()
+	for _, p := range activePods {
+		if p.UID != pod.UID {
+			otherActivePods = append(otherActivePods, p)
+		}
+	}
+	if ok, failReason, failMessage := kl.canAdmitPod(otherActivePods, pod); !ok {
+		klog.Warningf("Pod '%s' resize cannot be accommodated. Reason: '%s' Message: '%s'", pod.Name, failReason, failMessage) //TODO: logV(4) not warn
+		// TODO: Setup retry-backoff timer
+		return
+	}
+
+	for i, container := range pod.Spec.Containers {
+		for rName, rQuantity := range container.Resources.Requests {
+			pod.Spec.Containers[i].ResourcesAllocated[rName] = rQuantity
+		}
+	}
+	_, updateError := kl.kubeClient.CoreV1().Pods(pod.Namespace).Update(pod)
```

Comment:
It looks like this only works with pods in system space. Does this work with pods in tenant spaces?

Reply:
Thanks for catching this, I'll fix it.